### DISCUSSION TITLE
How do I know whether the selected entry is a method/function?

### DISCUSSION THREAD

**Discussion by adambasis** [hrsh7th/nvim-cmp#1473] [Q&A] [answered]
Hi,
I'm having a problem trying to get the type of the active entry.
When I use cmp.get_active_entry() or cmp.get_selected_entry() and I vim.inspect() it, then it prints not only **all** completion items, but also every single possible completion item with any letter.
I don't know if this is a bug, at least sounds like one.

Am I trying the correct thing? Maybe I am doing something wrong. I'd appreciate any help.
Thanks.

**Comment by fitrh** [comment, marked as answer]
> Am I trying the correct thing? 

What are you trying? What does your implementation look like? How do you use `cmp.get_{active,selected}_entry()`?

**Reply by adambasis** [reply]
I'm literally just opening the cmp window, selecting an item and call cmp.get_active_entry() through a mapping. When I print the result, it prints what I said in the original post.

**Reply by fitrh** [reply]
It would be better if you could share your code, for me, something like this was enough to get the entry kind

```lua
    ["<C-n>"] = cmp.mapping({
      i = function()
        if cmp.visible() then
          cmp.select_next_item()
          local entry = cmp.get_selected_entry()
          print(require("cmp.types").lsp.CompletionItemKind[entry:get_kind()])
        else
          cmp.complete()
        end
      end,
```

**Reply by adambasis** [reply]
Weird. This works for me too, but before I got to that point, I wrote print(vim.notify(entry)) and it printed out some random stuff, which it still does. Yet it works.
Thanks for your help.
